### ISSUE TITLE
Improve multi-input function interfaces

### ISSUE THREAD

**Issue by d21211122** [gchq/Gaffer#93]

**Comment by gaffer01** [comment]
Can a description of this issue be added please? It doesn't say anything about what the proposed improvements are.

**Comment by p013570** [comment]
Merged into develop-0.7.0

**Comment by p013570** [comment]
The functions library in Gaffer has been split into a separate project called [koryphe](https://github.com/gchq/koryphe). This has caused a large number of functions (filters, aggregators and transforms) to be moved and hence the package names have changed.

We have also aligned with Java 8 function interfaces:
```
FilterFunction -> Predicate
AggregateFunction -> BinaryOperator
TransformFunction -> Function
```
The interfaces for these methods has changed too.
```
FilterFunction.isValid -> Predicate.test
AggregateFunction.aggregate -> BinaryOperator.apply
SimpleAggregateFunction._aggregate -> KorypheBinaryOperator._apply
TransformFunction.transform -> Function.apply
```

this has also caused package names to be changed so match the java 8 packages.
```

uk.gov.gchq.gaffer.function.aggregate.FreqMapAggregator -> uk.gov.gchq.gaffer.binaryoperator.FreqMapAggregator
uk.gov.gchq.gaffer.function.aggregate.IntegerFreqMapAggregator -> uk.gov.gchq.gaffer.binaryoperator.IntegerFreqMapAggregator

uk.gov.gchq.gaffer.function.aggregate -> uk.gov.gchq.koryphe.impl.binaryoperator
uk.gov.gchq.gaffer.function.MapFilter -> uk.gov.koryphe.predicate.PredicateMap,
uk.gov.gchq.gaffer.function.filter -> uk.gov.gchq.koryphe.impl.predicate
uk.gov.gchq.gaffer.function.transform -> uk.gov.gchq.koryphe.impl.function
```
